src/transformers/models/auto/auto_factory.py

```diff
@@ -13,16 +13,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Factory function to build auto-model classes."""
+from __future__ import annotations
+
 import copy
 import importlib
 from collections import OrderedDict
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, Any, Tuple, Type, Union
 
 from ...configuration_utils import PretrainedConfig
 from ...dynamic_module_utils import get_class_from_dynamic_module
 from ...utils import copy_func, logging
 from .configuration_auto import AutoConfig, model_type_to_module_name, replace_list_option_in_docstrings
 
 
+if TYPE_CHECKING:
+    # Avoid runtime dependency on other modules for typing needs
+    from ...tokenization_utils_fast import PreTrainedTokenizerFast
+
+    _LazyAutoMappingValue = Tuple[
+        # Tokenizers will depend on packages installed, too much variance and there are no common base or Protocol
+        Type[Union[Any, None]],
+        Type[Union[PreTrainedTokenizerFast, None]],
```

Comment:
The value can be a tokenizer, a config or a model, so the type annotation is wrong here.